packages/core/src/node/process/process.ts

```diff
@@ -20,6 +21,7 @@ export interface IProcessExitEvent {
 @injectable()
 export abstract class Process implements Disposable {
 
+    readonly id: number;
     abstract readonly type: 'Raw' | 'Terminal';
     abstract pid: number;
     killed = false;
```

Comment:
`killed` should be `protected`? Otherwise, it can be reset by the client code?

Reply:
I want to keep this public but I'll protect the internal state with accessors so that it's read only.